### DISCUSSION TITLE
What is the browser compatibility like for the current 3.x version?

### DISCUSSION THREAD

**Discussion by YoringChen** [maplibre/maplibre-gl-js#3450] [Q&A] [answered]
Currently, I am experiencing some issues with Desktop Chrome 67.

Console Log for **maplibre-gl@3.3.1**:
`TypeError: Failed to execute 'uniformMatrix4fv' on 'WebGL2RenderingContext': No function was found that matched the provided signature.`

![697998011665801108](https://github.com/maplibre/maplibre-gl-js/assets/14165231/c8b1a9d8-6584-4ecf-b5e3-48dccd817123)

Console Log for **maplibre-gl@3.6.2**:
`TypeError: Failed to execute 'postMessage' on 'Worker': Iterator getter is not callable.`

![6974791961667579260](https://github.com/maplibre/maplibre-gl-js/assets/14165231/f419a277-c2ab-407d-8f5b-568ef22395fe)

Is Chrome 67 not within the compatibility range of v3.x? If there is any clear indication of browser compatibility available, kindly let me know.

Steps to reproduce the issue:

1. Download the [demo](https://github.com/YoringChen/maplibre-gl-test)
2. Run `npm install` -> `npm run build` -> `npm run preview`.
3. Open the URL http://localhost:4173/ to reproduce the issue."

**Comment by HarelM** [comment, marked as answer]
You can look at the relevant tag and check the tsconfig.json file to see which ES version the output is and compare it to the browser version you need to support.
In general, chrome 67 is a 5-year-old browser with security issues and we don't support it in newer versions.

**Reply by YohanSciubukgian** [reply]
@HarelM Do you have a documentation/reference on which browser versions are currently supported by the library?

**Reply by HarelM** [reply]
I don't unfortunately... Sorry...

**Reply by YohanSciubukgian** [reply]
Thanks for your answer.

AFAIU, the project has the following constraints (non-exhaustive list):
- WebGL2 with fallback to WebGL1 - [Release v3.1.0 · maplibre/maplibre-gl-js (github.com)](https://github.com/maplibre/maplibre-gl-js/releases/tag/v3.1.0)
Chrome 56 looks to be the first version of chrome that support all WebGL2 features. 
["webgl2" | Can I use... Support tables for HTML5, CSS3, etc](https://caniuse.com/?search=webgl2)
![image](https://github.com/maplibre/maplibre-gl-js/assets/30290851/370f3ab5-2222-4e23-8511-52b278c846d3)

- The TypeScript config that target "ES2019"
Chrome 73 looks to be the first version of chrome that support all ES2019 features.
Could we use polyfills to target older browser versions up to Chrome 56? if yes, could you provide some documentation guidance, using browserlist maybe?

Even if there is no guarantee as of today, it could help to have high level overview of browser support compatibility matrix (with or without polyfills) to engage on a business project that need minimal browser version support.
Could we improve the documentation to add a section that talk about browser compatibility support and requirements?
Could we have dedicated release notes highlight (or doing it at major version only) when newer APIs are being used that could bump the minimal browser version to avoid breaking business usages?

Side note, as you might consider using "ResizeObserver" ([Example check-for-support is outdated · Issue #3857 · maplibre/maplibre-gl-js (github.com)](https://github.com/maplibre/maplibre-gl-js/issues/3857)), it might bump the minimal chrome version support to Chrome 84 at minimal version based on ["ResizeObserver" | Can I use... Support tables for HTML5, CSS3, etc](https://caniuse.com/?search=ResizeObserver). It could be great to group this kind of API upgrade via a major version and have dedicated communication on new browser minimal baseline.
![image](https://github.com/maplibre/maplibre-gl-js/assets/30290851/0082be41-fe9b-48e2-a8f3-a38c2c4eb9d3)

_*I'm only talking about chrome versions just to simplify observations on API compatibility._

**Reply by YohanSciubukgian** [reply]
@HarelM Could you consider having a minimum browser version support as part of the major v5 upgrade ?
for example, it could be all versions where WebGL2 as you are planning to support it as a baseline in the future as per this discussion https://github.com/maplibre/maplibre-gl-js/issues/3834#issuecomment-2270169908).

**Reply by HarelM** [reply]
I, personally, have no intentions on tracking browsers compatibility. Latest version of maplibre-gl is targeting the latest browsers according to when it is released.
If your project needs old browsers (which is extremely not advised due to security issues and vulnerabilities) you can either use old versions of maplibre or test if the browser is supported. 
If you wish to track it and document it somewhere, your are very much welcome to do so.
